torch/_inductor/kernel/mm_scaled.py

```diff
@@ -584,3 +591,15 @@ def tuned_scaled_mm(
             "All choices for scaled_mm were invalid, using ATen backend as fallback"
         )
         return aten_choice.output_node()
+
+
+@functools.lru_cache
+def using_b200() -> bool:
+    """Returns true if the device is a NVIDIA B200, otherwise returns false."""
+    if not torch.cuda.is_available():
+        return False
+    for i in range(torch.cuda.device_count()):
+        device_name = torch.cuda.get_device_name(i)
+        if "B200" in device_name.upper():
```

Comment:
One more nit I would just get sm tags and ensure that is 10

`torch.cuda.get_device_properties(torch.cuda.current_device()).major`

Reply:
Done